packages/salesforcedx-vscode-apex/src/languageUtils/languageServerUtils.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { execSync } from 'child_process';
+import { SIGKILL } from 'constants';
+import { UBER_JAR_NAME } from '../constants';
+
+export type ProcessDetail = {
+  pid: number;
+  ppid: number;
+  command: string;
+  orphaned: boolean;
+};
+
+export function findAndCheckOrphanedProcesses(): ProcessDetail[] {
+  const platform = process.platform.toLowerCase();
+  const isWindows = platform === 'win32';
+
+  const cmd = isWindows
+    ? `Get-CimInstance -ClassName Win32_Process | Where-Object { $_.CommandLine -like '*${UBER_JAR_NAME}*' } | ForEach-Object { [PSCustomObject]@{ ProcessId = $_.ProcessId; ParentProcessId = $_.ParentProcessId; CommandLine = $_.CommandLine } } | Format-Table -HideTableHeaders`
+    : `ps -e -o pid,ppid,command | grep "${UBER_JAR_NAME}"`;
+
+  const stdout = execSync(
+    isWindows ? `powershell.exe -command "${cmd}"` : cmd
+  ).toString();
+  const lines = stdout.trim().split('\n');
+  const processes: ProcessDetail[] = lines.map(line => {
+    const [pidStr, ppidStr, ...commandParts] = line.trim().split(/\s+/);
+    const pid = parseInt(pidStr, 10);
+    const ppid = parseInt(ppidStr, 10);
+    const command = commandParts.join(' ');
+    return { pid, ppid, command, orphaned: false };
+  });
+
+  if (processes.length === 0) {
+    return [];
+  }
+
+  // Filter orphaned processes
+  const orphanedProcesses: ProcessDetail[] = processes
+    .map(processInfo => {
+      const checkOrphanedCmd = isWindows
```

Comment:
command to check if a process is orphaned. This is necessary given multiple language servers can be running and not be orphaned.

Reply:
I need to look at what this is doing, I should be checking if the reported parent process is still present.

Reply:
Yep, changing to use parent process Id